mongo/datadog_checks/mongo/mongo.py

```diff
@@ -175,9 +175,18 @@ def _build_metric_list_to_collect(self):
 
         return metrics_to_collect
 
+    def _refresh_replica_role(self):
+        if (
+            self._config.refresh_role
+            and self._api_client
+            and isinstance(self._api_client.deployment_type, ReplicaSetDeployment)
+        ):
+            self._api_client.deployment_type = self._api_client._get_deployment_type()
+
     def check(self, _):
         try:
             self._check()
+            self._refresh_replica_role()
```

Comment:
Should the refresh be before check?

Reply:
Good idea, I moved it to during the check run